File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-commons/src/components/screenStates.ts
export interface ScreenState {
  readonly kind: "loading" | "empty" | "error" | "ready";
  readonly message: string;
}

export function loadingState(): ScreenState {
  return { kind: "loading", message: "Loading" };
}

export function emptyState(message: string): ScreenState {
  return { kind: "empty", message };
}

export function errorState(message: string): ScreenState {
  return { kind: "error", message };
}

export function readyState(): ScreenState {
  return { kind: "ready", message: "" };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_commons/lib/src/widgets/screen_states.dart
import 'package:flutter/material.dart';

/// Standard screen state kinds shared by every capability screen.
enum AppstoreScreenStateKind { loading, empty, error, ready }

class AppstoreScreenState extends StatelessWidget {
  const AppstoreScreenState({
    required this.kind,
    required this.message,
    super.key,
  });

  final AppstoreScreenStateKind kind;
  final String message;

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(24),
        child: Text(message, textAlign: TextAlign.center),
      ),
    );
  }
}
